### ISSUE TITLE
No alert or confirm dialogs in iframes

### ISSUE THREAD

**Issue by etiktin** [electron/electron#2644]
Currently there's no support for `alert` and `confirm` dialogs in iframes (see discussion in #2522).

Once the "Session wide preload script" #2605 is ready, I think we should implement a default preload that will add the missing dialogs (kind of like what override.coffee does for the parent window).

**Comment by xland** [comment]
How is this progressing?

**Comment by qinyang912** [comment]
also on version 1.0

**Comment by r1si** [comment]
Any news? this is big bug

**Comment by r1si** [comment]
Ok, I have a valid workaround...
onload of iframe launch this code:
document.getElementById(ID_CURRENT_IFRAME).contentWindow.alert=alert;
document.getElementById(ID_CURRENT_IFRAME).contentWindow.confirm=confirm;

**Comment by cuixiping** [comment]
Just use this js code in iframe (no CORS):
```
window.alert = top.alert;
window.confirm = top.confirm;
window.prompt = top.prompt;
```

then use alert,confirm,prompt dialog as usual.

**Comment by razzbee** [comment]
Google disabled it for webview , but I used this to make it work, Note that I am using nw.js so there might be some slight difference, you must figure it out and make it work for you : 
```js
.addEventListener('dialog',function(e){
   
   //message type 
   messageType = e.messageType;

   messageText = e.messageText;

   DialogController = e.dialog;


   //lets checki if alert
   if(messageType == 'alert'){
     window.alert(messageText);
   }//emd if 

   //if confirm
   else if(messageType == 'confirm'){

    //confirm
    var confirm =  window.confirm(messageText);
   
    //get confirm bool and send to controller
    if(confirm){

       //if true send okay to browser
       DialogController.ok();

    }else{

      //send cancel with to send false false
      DialogController.cancel();
    }//end if

   }//end if confirm
  
  //lastly if its prompt 
  else if(messageType == 'prompt'){
     
     //get user Input 
     promptInput = window.prompt(messageText);

     //if null , then means the user clicked cancel 
     if(promptInput == null){

          //tell browser to cancel 
          DialogController.cancel();
     }else{
         //feed browser with input data 
         DialogController.ok(promptInput);
     }
  }//end if prompt
 
});//end dialog test 

```

**Comment by ydogandjiev** [comment]
@zcbenz, @cuixiping

Hey guys, is there any update on this issue? Having window.open return undefined prevents postMessage communication with child windows. I have yet to find a workaround.

**Comment by kevinsawicki** [comment]
`alert` and `confirm` are now available in `<iframe>` tags via #8341

Closing this out, there is still https://github.com/electron/electron/issues/2605 for a session wide preload script.

**Comment by poppinlp** [comment]
@kevinsawicki  Hi~

I find some issues about `window.open` in iframe will return `null` which has been closed and commend "merging to #2644".
And this issue is closed.

But that problem still could happen. You could see detail in [this issue](https://github.com/electron/electron/issues/9581) which has been opened 2 days ago.
And that problem hits me either.

**Comment by mojoaxel** [comment]
`window.propmt` is not supported in electron. see #472
